clients/drcachesim/drpt2trace/pt2ir.h

```diff
@@ -0,0 +1,263 @@
+
+        /* The MTC frequency. */
+        uint8_t mtc_freq;
+
+        /* The nominal frequency. */
+        uint8_t nom_freq;
+    } pt_config; /**< The libipt config of PT raw trace. We can get these parameters by
+                  *   running libipt/scirpts/perf-get-opts.bash.
+                  */
+
+    std::string raw_file_path; /**< The PT raw trace file path. */
+
+    /* The class pt2ir_t does not want to expose libipt-sb to the upper layer too. So we
+     * redefine some configurations of libipt-sb in sb_config. The libipt-sb sideband
+     * session requires these parameters. We can get these parameters by running
+     * libipt/scirpts/perf-get-opts.bash.
+     */
+    struct {
+        /* perf_event_attr.sample_type */
+        uint64_t sample_type;
+
+        /* The start address of kernel. The sideband session use it to distinguish kernel
+         * from user addresses: kernel >= @kernel_start user   <  @kernel_start
+         */
+        uint64_t kernel_start;
+
+        /* The sysroot is used for remote trace decoding. If the image locates at
+         * /path/to/image in remote machine, it will load it from ${sysroot}/path/to/image
+         * in local machine.
+         */
+        std::string sysroot;
+
+        /* The values of time_shift, time_mult, and time_zero are used to synchronize
+         * trace time, and the sideband recodes time.
+         *
+         *   time_shift = perf_event_mmap_page.ttime_shift
+         *   time_mult = perf_event_mmap_page.time_mult
+         *   time_zero = perf_event_mmap_page.time_zero
+         */
+        uint16_t time_shift;
+        uint32_t time_mult;
+        uint64_t time_zero;
+
+        /* If the trace contains coarse timing information, the sideband events may be
+         * applied too late to cause "no memory mmapped at this address" issue. To set the
+         * tsc_offset can fix the issue.
+         * XXX: The libipt documentation said we need to set a suitable value for
+         * tsc_offset. But it didn't give a method to get a suitable value. So we may need
+         * to find a way to generate the suitable tsc_offset.
+         */
+        uint64_t tsc_offset;
+    } sb_config; /**< The libipt-sb config of PT raw trace. We can get these parameters by
+                  *   running libipt/scirpts/perf-get-opts.bash.
+                  */
+
+    /* sb_primary_file_path is the path of the primary sideband file.
+     * sb_secondary_file_path_list is a list of paths to secondary sideband files.
+     * The sideband primary file and secondary files are stored perf event records. A
+     * primary sideband file contain perf event records for the traced cpu.  A secondary
+     * sideband contain perf event records for other cpus on the system.
+     */
+    std::string sb_primary_file_path; /**< The sideband primary file of PT raw trace. */
+    std::vector<std::string>
+        sb_secondary_file_path_list; /**< The sideband secondary file of PT raw trace. */
```

Comment:
These can be pre-field Doxygen comments if there's not enough space at the end of the line to fit easily as a post-field comment.

Reply:
Done.